tyssue/core/sheet.py

```diff
@@ -273,6 +275,75 @@ def extract_bounding_box(
         subsheet.reset_topo()
         return subsheet
 
+    def get_force_inference(self, coords=None):
+        """Measure force based on Brodland method.
+
+        g_gamma_matrix*tension_vector = 0
+        Equation is homogenous and to avoid tension_vectors = 0,
+        Construction and solve the constrained least-squares equation system
+        [[g_gamma_matrix.T g_gamma_matrix, C^T_1],
+         [C_1, 0]]
+         où C1 = {1....1}
+
+        """
+        if coords is None:
+            coords = self.coords
+        ndim = self.dim
+        dcoords = ["d" + c for c in coords]
+
+        self.get_extra_indices()
+        edges_index = self.east_edges
+        edges_index_opposite = self.west_edges
+
+        n_edges = len(edges_index)
+
+        srce, trgt = self.edge_df.loc[
+            edges_index, ['srce', 'trgt']].iteritems()
+        srce = srce[1].to_numpy()
+        trgt = trgt[1].to_numpy()
+
+        # Fill gamma matrix to measure tension
+        pos = (self.edge_df.loc[edges_index, dcoords].to_numpy() / np.repeat(np.linalg.norm(
+            self.edge_df.loc[edges_index, dcoords].to_numpy(), axis=1), ndim).reshape((n_edges, ndim)))
+
+        pos = np.concatenate((pos, -pos)).flatten()
+        row = np.concatenate((
+            np.array([np.arange(i * ndim, i * ndim + ndim)
+                      for i in srce]).flatten(),
+            np.array([np.arange(i * ndim, i * ndim + ndim)
+                      for i in trgt]).flatten())
+        )
+
+        col = np.concatenate(
+            (np.repeat(np.arange(n_edges), ndim), np.repeat(np.arange(n_edges), ndim)))
+
+        g_gamma_matrix = coo_matrix((pos, (col, row)), shape=(
+            n_edges, self.Nv * ndim)).toarray()
+
+        g_gamma_matrix = g_gamma_matrix.T
+
+        p = np.ones((n_edges + 1, n_edges + 1))
+
+        # get g_gamma_matrix.T g_gamma_matrix
+        g_gamma_matrix_dot = np.dot(g_gamma_matrix.T, g_gamma_matrix)
+        p[0:n_edges, 0:n_edges] = g_gamma_matrix_dot
+        p[n_edges, n_edges] = 0
+
+        q = np.zeros((n_edges + 1, 1))
+        q[n_edges, 0] = n_edges
+
+        # Compute QR decomposition of a matrix.
+        r1, r2 = linalg.qr(p)
+        y = np.dot(r1.T, q)
+
+        x = linalg.solve(r2, y)
+
+        tension = x[0:n_edges][:, 0]
+
+        self.edge_df['tension'] = np.nan
+        self.edge_df.loc[edges_index, 'tension'] = tension
+        self.edge_df.loc[edges_index_opposite, 'tension'] = tension
```

Comment:
maybe it's better to return the column and let the user assign it (or, alternatively set the column name as a parameter in the function call)